WordPress/Classes/ViewRelated/Aztec/ViewControllers/AztecPostViewController.swift

```diff
@@ -2827,17 +2827,17 @@ extension AztecPostViewController {
            videoSrcURL.scheme == VideoShortcodeProcessor.videoPressScheme,
            let videoPressID = videoSrcURL.host {
             // It's videoPress video so let's fetch the information for the video
-            let mediaService = MediaService(managedObjectContext: ContextManager.sharedInstance().mainContext)
-            mediaService.getMetadataFromVideoPressID(videoPressID, in: self.post.blog, success: { (metadata) in
-                if let originalURL = metadata.originalURL {
+            let remote = MediaServiceRemoteFactory().remote(for: self.post.blog)
+            remote?.getMetadataFromVideoPressID(videoPressID, isSitePrivate: self.post.blog.isPrivate(), success: { metadata in
+                if let metadata, let originalURL = metadata.originalURL {
                     videoAttachment.updateURL(metadata.getURLWithToken(url: originalURL) ?? originalURL)
                 }
-                if let posterURL = metadata.posterURL {
+                if let metadata, let posterURL = metadata.posterURL {
                     videoAttachment.posterURL = metadata.getURLWithToken(url: posterURL) ?? posterURL
```

Comment:
It took me a while to understand why `let metadata` was defined "twice". 

This seems like a bit of an odd pattern `metadata.getURLWithToken(url: originalURL) ?? originalURL`, but not much to do about it here 🤔